tests/unit/test_lineage.py

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from unittest import mock
+
+import pytest
+
+from snowflake.snowpark.exceptions import SnowparkFetchDataException
+from snowflake.snowpark.lineage import (
+    Lineage,
+    _ObjectField,
+    _SnowflakeDomain,
+    _UserDomain,
+)
+from snowflake.snowpark.session import Session
+
+
+def test_get_name_and_version():
+    fake_session = mock.create_autospec(Session, _session_id=123456)
+    fake_session._analyzer = mock.Mock()
+
+    graph_entity = {
+        _ObjectField.USER_DOMAIN: _UserDomain.MODEL,
+        _ObjectField.DB: "db1",
+        _ObjectField.SCHEMA: "schema1",
+        _ObjectField.PROPERTIES: {_ObjectField.PARENT_NAME: "name1"},
+        _ObjectField.NAME: "version1",
+    }
+    name, version = Lineage(fake_session)._get_name_and_version(graph_entity)
+    print(name, version)
```

Comment:
nit: maybe get rid of this print?

Reply:
ahh. Running against time :) I will fix this on a follow up PR.